### ISSUE TITLE
Performance issues in relational queries

### ISSUE THREAD

**Issue by HonQii** [realm/realm-cocoa#7396]
I have three tables; among them, table A is the material, with more than 7000 elements; table C is the attribute of the material, with more than 50 elements; table B is the association table between A and C, because there are also between A and C Quantity and volume are attributes, so I use table B to make associations and record these additional attributes, instead of directly using the array that stores C in table A.
Now when I need to display the list of A and its associated attributes, it will be very stuck; I think it is because Realm uses a delayed query, which causes me to display an element and its attributes, which need to be queried hundreds of times (first query table B, Look up table C)

## Goals
Can complete the queries I need with high performance

## Expected Results
1. Is it possible to query all related tables through one query like sql, instead of delaying multiple queries
2. Or is there any good way to solve the performance problem

## Actual Results
Every time the page is displayed, it will be stuck for ten seconds or even dozens of seconds

## Code Sample
```swift
class TableA {
  var name: string = ""
  var length: Float = 0
  var width: Float = 0
  .....
  let linkInfos = LinkingObjects(fromType: TableB.self, property: "tableA")
}

class TableB {
  var volume: Float = 0
  var amount: Float = 0
  .....
  var tableA: TableA?
  var tableC: TableC?
}

class TableC {
  var propertyA: string = ""
  var propertyB: string = ""
}
```

## Steps for others to Reproduce
Query TableA and display it with tableView, read the data in TableB and TableC in tableViewCell

## Version of Realm and Tooling
Realm framework version: 10.10.0

Xcode version: 12.5.1

iOS/OSX version: 14.4.2

Dependency manager + version: SPM

**Comment by ericjordanmossman** [comment]
Are you currently displaying the values from TableB/C in the cell via the `linkInfos` property? Or are you running multiple queries?

**Comment by HonQii** [comment]
> Are you currently displaying the values from TableB/C in the cell via the `linkInfos` property? Or are you running multiple queries?

Yes, I display values from TableB and TableC via linkInfos property

**Comment by ericjordanmossman** [comment]
@HonQii 
Are you able to create a distilled example of the problem? It'd be nice to try a couple things to figure out what is happening. Like seeing how the query behaves after removing references to `linkInfos.someValue`.

**Comment by ericjordanmossman** [comment]
Also are you measuring the slowness via the UI only? Or do you see similar issues if measuring just the results of the queries?

**Comment by dianaafanador3** [comment]
@HonQii did you try @ericjordanmossman suggestion?
